planner/core/rule_eliminate_projection.go

```diff
@@ -137,6 +137,14 @@ func (pe *projectionEliminater) eliminate(p LogicalPlan, replace map[string]*exp
 		}
 	}
 	p.replaceExprColumns(replace)
+	if isProj {
+		if child, ok := p.Children()[0].(*LogicalProjection); ok {
+			for i := range proj.Exprs {
+				proj.Exprs[i] = replaceColumnOfExpr(proj.Exprs[i], child)
+			}
+			p.Children()[0] = child.Children()[0]
+		}
+	}
```

Comment:
Can we fix this issue by removing the constraint of `canProjectionBeEliminatedLoose` and enhancing `replace map[string]*expression.Column` to `replace map[string]expression.Expression`? we need to update `resolveExprAndReplace` though.

Reply:
Some operator will use `replace map[string]*expression.Column` to replace the schema.
I am not sure how to handle it.